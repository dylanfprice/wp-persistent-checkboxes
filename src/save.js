import md5 from 'md5';
import ReactDOM from 'react-dom';

import parseLabels from './parseLabels';
import PersistentCheckboxList from './PersistentCheckboxList';

export default function save( { attributes: { labelObjects }, className } ) {
	const labels = labelObjects.map( ( { label } ) => label );
	const script = (
		`window.addEventListener('load', function(event) {
       persistentCheckboxes.render(${ JSON.stringify( { labels } ) })
     });`
	);
	return (
		<div id={ getBlockId( labels ) } className={ className }>
			<PersistentCheckboxList labels={ labels } />
			<script dangerouslySetInnerHTML={ { __html: script } }></script>
		</div>
	);
}

export function getBlockId( labels ) {
	return `wp-persistent-checkboxes-${ md5( labels.join( '' ) ) }`;
}

export function render( { labels } ) {
	ReactDOM.render(
		<PersistentCheckboxList labels={ labels } />,
		document.getElementById( getBlockId( labels ) )
	);
}
