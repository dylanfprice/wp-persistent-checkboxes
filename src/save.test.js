import { shallow } from 'enzyme';

import save, * as module from './save';

describe( 'save', () => {
	it( 'renders the checkboxes and inline script', () => {
		const labelObjects = [
			{ label: 'label one' },
			{ label: 'label two' },
			{ label: 'label three' },
		];
		const saveContent = save( {
			attributes: { labelObjects },
			className: 'test',
		} );
		expect( shallow( saveContent ) ).toMatchSnapshot();
	} );
} );

describe( 'getBlockId', () => {
	it( 'returns block id for given labels', () => {
		const actual = module.getBlockId( [ 'label one', 'label two' ] );
		expect( actual ).toBe(
			'wp-persistent-checkboxes-de0e1e73c6e1e2bb1e5870f76a0e84b8'
		);
	} );
} );
