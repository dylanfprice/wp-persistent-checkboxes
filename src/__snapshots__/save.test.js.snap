// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`save renders the checkboxes and inline script 1`] = `
<div
  className="test"
  id="wp-persistent-checkboxes-f71ed41bc0cd46acf4084643477bdd31"
>
  <PersistentCheckboxList
    labels={
      Array [
        "label one",
        "label two",
        "label three",
      ]
    }
  />
  <script
    dangerouslySetInnerHTML={
      Object {
        "__html": "window.addEventListener('load', function(event) {
       persistentCheckboxes.render({\\"labels\\":[\\"label one\\",\\"label two\\",\\"label three\\"]})
     });",
      }
    }
  />
</div>
`;
